<a href="https://colab.research.google.com/github/alexander-billings/stroke_prediction/blob/main/Stroke_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("https://raw.githubusercontent.com/alexander-billings/stroke_prediction/main/stroke_data.csv")
df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


## Are there columns that should be dropped?

### None at a glance

## Are there duplicate columns?

In [5]:
df.duplicated().sum()

0

### No duplicate rows found

## Are there any rows with missing data?

In [7]:
df.isna().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

In [11]:
df.drop(df.loc[df['bmi'].isna()].index, inplace=True)

### Dropped rows with missing data
Considering it was only 201 rows with missing data in the bmi column, I went ahead and dropped it. I will return to this if balance starts to impact model performance.

## Are there any inconsistent categories?

In [22]:
for column in df.columns:
    if df[column].dtype == 'object':
        print("Values in", column)
        print(df[column].value_counts(), '\n')

Values in gender
Female    2897
Male      2011
Other        1
Name: gender, dtype: int64 

Values in ever_married
Yes    3204
No     1705
Name: ever_married, dtype: int64 

Values in work_type
Private          2811
Self-employed     775
children          671
Govt_job          630
Never_worked       22
Name: work_type, dtype: int64 

Values in Residence_type
Urban    2490
Rural    2419
Name: Residence_type, dtype: int64 

Values in smoking_status
never smoked       1852
Unknown            1483
formerly smoked     837
smokes              737
Name: smoking_status, dtype: int64 



### Category observations
* Gender: "Other" found in one row. There is not enough of a sample size to determine correlations here, might consider dropping said row.
* Smoking: Large number of "Unkown" smokers. Considering the medical literature around smoking increasing risk of stroke, I want to keep this column in. I have decided to leave "unkown" as-is, pending model performance.

In [13]:
df['stroke'].value_counts()

0    4700
1     209
Name: stroke, dtype: int64